Lib/test/test_minidom.py

```diff
@@ -395,14 +395,27 @@ def testChangeAttr(self):
                 and el.getAttribute("spam2") == "bam2")
         dom.unlink()
 
-    def testGetAttrList(self):
-        pass
+    def testGetAttrListAndLength(self):
+        dom = parseString("<abc/>")
+        el = dom.documentElement
+        el.setAttribute("spam", "jam")
+        self.assertEqual(len(el.attributes.items()), 1)
+        el.setAttribute("foo", "bar")
+        self.assertEqual(len(el.attributes.items()), 2)
+        self.assertIn(('spam', 'jam'), el.attributes.items())
+        self.assertIn(('foo', 'bar'), el.attributes.items())
+        dom.unlink()
```

Comment:
```suggestion
        dom = parseString("<abc/>")
        self.addCleanup(dom.unlink)
        el = dom.documentElement
        el.setAttribute("spam", "jam")
        self.assertEqual(len(el.attributes.items()), 1)
        el.setAttribute("foo", "bar")
        items = el.attributes.items()
        self.assertEqual(len(items), 2)
        self.assertIn(('spam', 'jam'), items)
        self.assertIn(('foo', 'bar'), items)
```

Reply:
Strictly speaking, we are testing the `.items()` method of `attributes`, which is a `NamedNodeMap` and not a `dict`. I don't know if we don't need to cover more methods of that or if this test is only for testing the content of `NamedNodeMap` without testing the rest of its API.